code/modules/paperwork/photocopier.dm

```diff
@@ -58,55 +98,75 @@
 		ass = null
 	if(deleting_atom == toner_cartridge)
 		toner_cartridge = null
+	if(deleting_atom in paper_stack)
+		paper_stack -= deleting_atom
 	return ..()
 
 /obj/machinery/photocopier/Destroy()
 	QDEL_NULL(object_copy)
 	QDEL_NULL(toner_cartridge)
+	QDEL_LIST(paper_stack)
 	ass = null //the mob isn't actually contained and just referenced, no need to delete it.
 	return ..()
 
+/obj/machinery/photocopier/examine(mob/user)
+	. = ..()
+	if(object_copy)
+		. += span_notice("There is something inside the scanner tray.")
+	. += span_boldnotice("You can put any type of blank paper inside to print a form or to copy something onto it.")
 
 /obj/machinery/photocopier/ui_interact(mob/user, datum/tgui/ui)
 	ui = SStgui.try_update_ui(user, src, ui)
 	if(!ui)
 		ui = new(user, src, "Photocopier")
 		ui.open()
 
+/obj/machinery/photocopier/ui_static_data(mob/user)
+	var/list/static_data = list()
+
+	var/list/blank_infos = list()
+	var/list/category_names = list()
+	if(GLOB.paper_blanks)
+		for(var/blank_id in GLOB.paper_blanks)
+			var/list/paper_blank = GLOB.paper_blanks[blank_id]
+			blank_infos += list(list(
+				name = paper_blank["name"],
+				category = paper_blank["category"],
+				code = blank_id
+			))
+			category_names |= paper_blank["category"]
+
+	static_data["blanks"] = blank_infos
+	static_data["categories"] = category_names
+
+	return static_data
+
 /obj/machinery/photocopier/ui_data(mob/user)
 	var/list/data = list()
 	data["has_item"] = !copier_empty()
 	data["num_copies"] = num_copies
 
-	try
-		var/list/blanks = json_decode(file2text("config/blanks.json"))
-		if (blanks != null)
-			data["blanks"] = blanks
-			data["category"] = category
-			data["forms_exist"] = TRUE
-		else
-			data["forms_exist"] = FALSE
-	catch()
-		data["forms_exist"] = FALSE
+	data["category"] = category
+	data["copies_left"] = copies_left
 
 	if(istype(object_copy, /obj/item/photo))
 		data["is_photo"] = TRUE
 		data["color_mode"] = color_mode
 
 	if(isAI(user))
 		data["isAI"] = TRUE
-		data["can_AI_print"] = toner_cartridge ? toner_cartridge.charges >= PHOTO_TONER_USE : FALSE
+		data["can_AI_print"] = (toner_cartridge ? toner_cartridge.charges >= PHOTO_TONER_USE : FALSE) && get_paper_count() >= PHOTO_PAPER_USE
```

Comment:
```suggestion
		data["can_AI_print"] = toner_cartridge && (toner_cartridge.charges >= PHOTO_TONER_USE) && (get_paper_count() >= PHOTO_PAPER_USE)
```

it reads a bit better to do it like this now that we're adding more stuff onto this line.